game-core/src/main/java/games/strategy/engine/chat/ChatMessagePanel.java

```diff
@@ -224,10 +224,10 @@ private void cleanupKeyMap() {
     nextMessageKeymap.remove(KeyStroke.getKeyStroke(KeyEvent.VK_DOWN, 0, false));
   }
 
-  /** thread safe. */
   @Override
-  public void messageReceived(final String message, final PlayerName from) {
-    addMessageWithSound(message, from, SoundPath.CLIP_CHAT_MESSAGE);
+  public void messageReceived(final ChatMessage chatMessage) {
+    addMessageWithSound(
+        chatMessage.getMessage(), chatMessage.getFrom(), SoundPath.CLIP_CHAT_MESSAGE);
```

Comment:
Here and in line 115:
Why isn't ChatMessage passed there as well?

Reply:
Same reasoning as previous, essentially a slap event will be coming along, not sure yet how to reconcile that with the 'addMessageWithSound'